lib/lists.rego

```diff
@@ -27,10 +27,7 @@ index_last(l, x) = t[minus(count(t), 1)] {
 }
 
 # Reverse provided list
-reverse(l) = [x |
-		i := numbers.range(count(l) - 1, 0)[_]
-		x := l[i]
-	]
+reverse(l) = [l[j] | _ = l[i]; j := count(l)-1-i]
```

Comment:
Bordering on code golf here :D But yeah it feels more "native" - I might re-add the new lines later though for readability.